python/datasources/cdc_hiv.py

```diff
@@ -94,144 +48,110 @@ def get_table_name():
-            missing_data), std_col.HIV_DIAGNOSES] = np.nan
-        combined_group_df[std_col.HIV_DIAGNOSES] = combined_group_df[std_col.HIV_DIAGNOSES].astype(
-            float)
-        combined_group_df[std_col.HIV_DIAGNOSES_PER_100K] = combined_group_df[std_col.HIV_DIAGNOSES_PER_100K].astype(
-            float)
-
-        combined_group_df = generate_pct_share_col_without_unknowns(
-            combined_group_df, pct_share_dict, breakdown, 'All')
+            std_col.HIV_POPULATION_PCT,
+            std_col.HIV_DIAGNOSES_PCT_INEQUITY]
+
+        breakdown_group_df = generate_df_from_data_dir(
+            f'hiv-{geo_level}-{breakdown}.csv')
+
+        combined_group_df = pd.concat([breakdown_group_df, alls_df], axis=0)
+
+        df = combined_group_df.rename(columns=columns_to_standard)
+
+        df = df.replace(to_replace=HIV_TERMS_STANDARD_BY_COL)
+
+        if geo_level == COUNTY_LEVEL:
+            df = merge_county_names(df)
+            df[std_col.STATE_FIPS_COL] = df[std_col.COUNTY_FIPS_COL].str.slice(0,
+                                                                               2)
+
+        if geo_level == NATIONAL_LEVEL:
+            df[std_col.STATE_FIPS_COL] = US_FIPS
 
         if breakdown == std_col.RACE_OR_HISPANIC_COL:
-            combined_group_df = combined_group_df.replace(
-                {std_col.RACE_OR_HISPANIC_COL: RACE_GROUPS_TO_STANDARD})
-            combined_group_df = combined_group_df.replace(['All'], 'ALL')
-            combined_group_df = combined_group_df.rename(
-                columns={std_col.RACE_OR_HISPANIC_COL: std_col.RACE_CATEGORY_ID_COL})
-            std_col.add_race_columns_from_category_id(combined_group_df)
-            cols.append(std_col.RACE_CATEGORY_ID_COL)
+            std_col.add_race_columns_from_category_id(df)
+            columns_to_keep.append(std_col.RACE_CATEGORY_ID_COL)
+
+        # replace string number with whole number
+        df[std_col.HIV_DIAGNOSES] = df[std_col.HIV_DIAGNOSES].replace(',',
+                                                                      '',
+                                                                      regex=True)
+
+        df = generate_pct_share_col_without_unknowns(df,
+                                                     PCT_SHARE_DICT,
+                                                     breakdown,
+                                                     std_col.ALL_VALUE)
+
+        df = generate_pct_rel_inequity_col(df,
+                                           std_col.HIV_DIAGNOSES_PCT_SHARE,
+                                           std_col.HIV_POPULATION_PCT,
+                                           std_col.HIV_DIAGNOSES_PCT_INEQUITY)
 
-        df = combined_group_df[cols]
+        df = df[columns_to_keep]
+        df = df.sort_values([FIPS, breakdown]).reset_index(drop=True)
 
         return df
+
+
+def generate_df_from_data_dir(filename: str):
```

Comment:
should this be `generate_` or should it be like `load_` ? it's not really generating any new figures, it's just loading in the csv as a df right? also I would consider passing in the geo and demo as 2 args and building the filename inside this function, rather than building the f string outside and sending it in... but that's just a preference thing.